### PR TITLE
Shooting State Machine

### PR DESCRIPTION
**Pull request by TrueDemoknight**

### CODE REVIEW

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -234,4 +272,158 @@ public void RobotLoop() {
...
+      case PREPARE_TO_SHOOT:
+
+        /** Starts roller */
+        mShooter.start();
+
+        //TODO: Placeholder method, replace later
+        mShooter.target();
+
+        /* If rollers are spun up, changes to next state */
+        if (mShooter.isAtVelocity()) {
```
Comment (jhinspeter) [inline src/main/java/frc/robot/Robot.java]: AND target acquired or put a TODO

File: src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -234,4 +272,158 @@ public void RobotLoop() {
...
+
+        /* If finished shooting, changes to next state*/
+        if (mShooter.isDoneShooting()) {
+          mShootingState = ShootingState.SHOOT_BALL_COMPLETE;
+        }
+        break;
+      case SHOOT_BALL_COMPLETE:
+        /* Decrements storage */
+        mStorage.removeBall();
+
```
Comment (jhinspeter) [inline src/main/java/frc/robot/Robot.java:460]: Need to turn the storage rollers OFF

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 3]
```diff
@@ -182,15 +92,36 @@ public void target() {
...
+  }
+
+  /** Returns whether the firing timer has run longer than the duration. */
+  public boolean isDoneShooting() {
+    if (mFireTimer.get() >= FIRE_DURATION_SECONDS * 1000) {
+      stop();
+      mFireTimer.stop();
+      mFireTimer.reset();
+    }
+    return mFireTimer.get() >= FIRE_DURATION_SECONDS * 1000;
```
Comment (jhinspeter) [inline src/main/java/frc/robot/subsystems/Shooter.java]: Why are you comparing twice? Instead use an else, and return false in the else part. Return true in the if.

**Review by jhinspeter — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (TrueDemoknight) [reply src/main/java/frc/robot/Robot.java]: TODONE

File: src/main/java/frc/robot/subsystems/Shooter.java (hunk 3 above)
Comment (TrueDemoknight) [reply src/main/java/frc/robot/subsystems/Shooter.java]: Resolved

File: src/main/java/frc/robot/Robot.java (hunk 2 above)
Comment (TrueDemoknight) [reply src/main/java/frc/robot/Robot.java:460]: Resolved

**Review by jhinspeter — APPROVED** [review]

**Pull request merged**